chia/data_layer/data_layer_wallet.py

```diff
@@ -220,27 +220,15 @@ async def track_new_launcher_id(
         if await self.wallet_state_manager.dl_store.get_launcher(launcher_id) is not None:
             self.log.info(f"Spend of launcher {launcher_id} has already been processed")
             return None
-        if spend is not None and spend.coin.name() == launcher_id:  # spend.coin.name() == launcher_id is a sanity check
-            await self.new_launcher_spend(spend, peer, height)
-        else:
+        if spend is None or height is None:
             launcher_state: CoinState = await self.get_launcher_coin_state(launcher_id, peer)
-            launcher_spend = await fetch_coin_spend_for_coin_state(launcher_state, peer)
-            await self.new_launcher_spend(launcher_spend, peer)
+            spend = await fetch_coin_spend_for_coin_state(launcher_state, peer)
+            assert launcher_state.spent_height is not None
```

Comment:
any thought to raising here?

Reply:
`assert` will throw an `AssertionFailed` exception. are you suggesting that this is an inappropriate exception because it might fail in some legitimate case?

Reply:
It doesn't matter much. I just think if something might happen we might as well raise something specific - I'm not much a fan of just "assert".

Reply:
In this case `fetch_coin_spend_for_coin_state` would anyway already raise if its `None`. Its to make mypy happy only.